lumeer-storage/lumeer-storage-mongodb/src/main/java/io/lumeer/storage/mongodb/dao/project/MongoCollectionDao.java

```diff
@@ -91,12 +91,13 @@ public Collection updateCollection(final String id, final Collection collection)
       FindOneAndReplaceOptions options = new FindOneAndReplaceOptions().returnDocument(ReturnDocument.AFTER);
 
       try {
-         Collection updatedCollection = databaseCollection().findOneAndReplace(idFilter(id), collection, options);
+         final Collection storedCollection = databaseCollection().find(idFilter(id)).first();
```

Comment:
Why you don't use collection from method parameter?

Reply:
Because it is already updated. There is no other way to get to the old value unless I want to make a copy on all places where the method is called :-(

Reply:
But why it works in other places (ProjectDao, OrganizationDao, ViewDao) without this?

Reply:
It does not, the original object is simply passed as the method parameter.

Reply:
In `MongoProjectDao` you can find `public Project updateProject(final String projectId, final Project project, final Project originalProject)`

Reply:
So you can add similar method for `CollectionDao`.

Reply:
I really tried at first. Unfortunately, in this case it is not that simple. Try to lookup for usages and see where the original Collection is available and where not. The CollectionDao is used on quite a lot of places and we would need to pass the original collection through several method calls typicaly. Do you think it is worth it? Would that be a better approach? Definitely more DB efficient. WDYT?